### PR TITLE
feat(AI): Add AI agent tooling for streamlined co-working 

### PR DESCRIPTION
**Pull request by Chocolate-lightning**
## Description

This PR adds comprehensive AI agent tooling to the repository, giving tools like Claude Code, GitHub Copilot, and other AI assistants the context they need to contribute safely and consistently to the design system.

Two changes are included:

**1. AI instruction files for GitHub Copilot / AI agents**

A root-level `copilot-instructions.md` covers repo shape, setup commands, commit conventions, testing workflow, and guardrails. Three path-scoped instruction files auto-load in VS Code when a matching file is open:

- `.github/instructions/components.instructions.md` — component authoring patterns, CSS conventions, prop types, and the component Definition of Done
- `.github/instructions/stories-tests.instructions.md` — story structure, interaction test patterns, a11y testing, and fuzz testing with `fast-check`
- `.github/instructions/tokens.instructions.md` — token pipeline rules and the ZeroHeight-managed read-only constraint on token sources

Key guardrails baked into the instructions prevent AI agents from editing token source files, adding unsanctioned npm dependencies, inventing ad-hoc CSS values instead of `--mds-*` tokens, or touching release automation and workflow files.

**2. Claude Code instructions (`.claude/CLAUDE.md`)**

Added a `.claude/CLAUDE.md` entry point that points Claude Code to the GitHub instruction files as the authoritative source of guidance, avoiding duplication.

## Related Jira or GitHub Issue

https://moodle.atlassian.net/browse/MDS-408

## Type of Change

- [x] Feature
- [ ] Bugfix
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Screenshots/Previews

No UI changes. No Storybook story changes.

## Checklist

- [x] I have updated, added, and run tests such as JSUnit, Storybook interactions, or fuzzing to prove my fix is effective or that my feature works
- [x] I have updated or added Storybook stories for new or changed components (if appropriate) — N/A, no component changes
- [x] I have considered accessibility and described any improvements or issues — N/A, no UI changes
- [x] I have considered security implications and referenced [SECURITY.md](./SECURITY.md) if relevant — instruction files explicitly guard against agents touching release automation, token sources, or workflow files

## Additional Notes

N/A

### CODE REVIEW

File: .github/instructions/components.instructions.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+// In the component body:
+const resolvedVariant =
+  variant && allowedVariants.includes(variant as ButtonVariant)
+    ? variant
+    : 'primary'; // fallback to default
+```
+
+## Class composition
+
+Build `className` as an array and join with a space. Always include the `mds-*` hook first, then the Bootstrap base class (e.g. `btn`), then Bootstrap modifier classes (e.g. `btn-primary`), then any consumer-provided `className`. Both the `mds-*` and Bootstrap classes are required — Bootstrap provides the base styles, `mds-*` is the hook for MDS token overrides:
```
Comment (andrewgos) [inline .github/instructions/components.instructions.md]: i think this one is too narrow as an instruction? we might have a different approach for different components. For example, we wouldn't need it when a component is not extending Bootstrap as another base class

File: .github/instructions/stories-tests.instructions.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+applyTo: 'components/**/*.{stories,test}.tsx,tests/**/*.ts'
+---
+
+# Stories and Tests Instructions
+
+## Story files (CSF format)
+
+### Meta object
+
+```ts
```
Comment (andrewgos) [inline .github/instructions/stories-tests.instructions.md]: are these one works like a code example? cant we just point to re-use similar format like the existing ones?

File: .github/instructions/stories-tests.instructions.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  },
+} satisfies Meta<typeof Button>;
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+```
+
+### Story definitions
+
+```ts
```
Comment (andrewgos) [inline .github/instructions/stories-tests.instructions.md]: same here. are these just code examples?

File: .github/instructions/stories-tests.instructions.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+### a11y
+
+Every story tagged `test` is automatically scanned by Axe against WCAG 2.x AA and best-practice rules (configured in `.storybook/preview.ts`). Violations are reported as errors. Do not disable a11y checks without a documented reason.
+
+## Unit tests (Vitest + jsdom)
+
+Run with `npm run test-unit`. Tests live in `ComponentName.test.tsx` alongside the component.
+
+### What to cover
```
Comment (andrewgos) [inline .github/instructions/stories-tests.instructions.md:45]: if we do this, will it limits its exploration thinking suggesting more edge cases?

File: .github/instructions/stories-tests.instructions.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+- `mds-*` class applied to root element
+- Correct Bootstrap classes applied for each variant and size
+- Invalid variant prop falls back to the default
+- Extra props forwarded to the underlying element (`data-testid`, `aria-label`, etc.)
+- Disabled state behaves correctly
+- Label renders as text content
+
+### Example
```
Comment (andrewgos) [inline .github/instructions/stories-tests.instructions.md]: Whats the use of this example? maybe more description on what to copy/follow from here?

similar to the previous comment, can it just point to another file for reference/example?

File: .github/instructions/stories-tests.instructions.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+});
+```
+
+### Property-based fuzzing
+
+Use fuzzing only when a component has a large combinatorial prop space where exhaustive manual cases would be impractical (e.g. 3+ independent props each with multiple values). For simple components with a small number of variants, standard `it` tests are sufficient and preferred.
+
+Use `fuzzComponent` from `tests/utils/fuzzComponent.ts` with `fast-check` when a component has many prop combinations. It renders the component with randomly generated props and asserts the key text is visible.
+
+```ts
```
Comment (andrewgos) [inline .github/instructions/stories-tests.instructions.md]: similar to the previous comment, can it just point to another file for reference/example?

File: .github/instructions/tokens.instructions.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+# Tokens Instructions
+
+## Agent guardrail — do not modify token files
+
+**Agents must not edit any token files.** `tokens/dtcg/**/*.json`, `tokens/css/**`, and `tokens/scss/**` are all managed externally:
+
+- `tokens/dtcg/*.json` — governed by an automated ZeroHeight PR flow. Changes must go through that process, not be made locally by an agent.
+- `tokens/css/*` and `tokens/scss/*` — generated outputs rebuilt from DTCG sources. Never edit directly.
+
+If a token change is needed, direct the contributor to request it via the Moodle Design System feedback form at https://moodle.zeroheight.com/.
```
Comment (andrewgos) [inline .github/instructions/tokens.instructions.md]: i believe this should be https://design.moodle.com/ ?

Comment (andrewgos) [reply .github/instructions/tokens.instructions.md]: would GitHub issue be more appropriate here? 🤔

File: .github/instructions/tokens.instructions.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+## Discovering available tokens
+
+Browse the generated files in `tokens/css/` (e.g. `colors.css`, `spacing.css`, `typography.css`) or use the Figma MCP token definitions via `get_design_context` or `get_variable_defs` against the Moodle Design System Figma team files.
+
+## Using tokens in component CSS
+
+Use CSS custom properties via `var()`. Never hardcode color, spacing, typography, border, or shadow values. Do not add fallback values to `var()` calls — tokens are always defined via the `index.ts` import chain, so a fallback would only mask a missing token that should be fixed at the source.
+
+```css
```
Comment (andrewgos) [inline .github/instructions/tokens.instructions.md]: would examples be necessary here? 🤔

File: .github/instructions/tokens.instructions.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+## Breaking change guardrail
+
+Renaming or removing a token is a breaking change — consumer code referencing the old name will silently lose its value (the `var()` resolves to nothing with no error).
+
+The following changes must not be made without a major version bump:
+
+| Change | Why it breaks |
+|---|---|
+| Renaming a token in `tokens/dtcg/` | Any `var(--mds-old-name)` in consumer CSS stops resolving |
```
Comment (andrewgos) [inline .github/instructions/tokens.instructions.md]: i dont think it's specifically needed to be under `tokens/dtcg/`. The rule applies to any tokens

File: .github/copilot-instructions.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+- Commits must follow [Conventional Commits](https://www.conventionalcommits.org/).
+- Allowed types: `build`, `chore`, `ci`, `docs`, `feat`, `fix`, `perf`, `refactor`, `revert`, `style`.
+- Subject must use sentence-case.
+
+## Guardrails
+
+**When adding any CSS value (color, spacing, typography, border, shadow):**
+1. Check `tokens/css/` or use Figma MCP (`get_variable_defs` / `get_design_context`) to find an existing `--mds-*` token.
+2. If a matching token exists → use it via `var(--mds-*)`.
+3. If no token exists → do not invent an ad-hoc value. Ask contributors to request one at https://moodle.zeroheight.com/.
```
Comment (andrewgos) [inline .github/copilot-instructions.md]: design.moodle.com ?

File: .github/instructions/tokens.instructions.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+**Agents must not edit any token files.** `tokens/dtcg/**/*.json`, `tokens/css/**`, and `tokens/scss/**` are all managed externally:
+
+- `tokens/dtcg/*.json` — governed by an automated ZeroHeight PR flow. Changes must go through that process, not be made locally by an agent.
+- `tokens/css/*` and `tokens/scss/*` — generated outputs rebuilt from DTCG sources. Never edit directly.
+
+If a token change is needed, direct the contributor to request it via the Moodle Design System feedback form at https://moodle.zeroheight.com/.
+
+## Generated files — do not hand-edit
+
+`tokens/css/*` and `tokens/scss/*` are generated outputs. They are overwritten every time `npm run build-tokens` runs. Never edit them directly.
```
Comment (andrewgos) [inline .github/instructions/tokens.instructions.md]: there seems to be multiple notes of this across various instructions files, perhaps we can tidy it up?

**Review by andrewgos — CHANGES_REQUESTED** [review]
here are few thoughts i have while manually reading it through. will try to test it out and let you know if i find something else

File: .github/instructions/components.instructions.md (hunk 1 above)
Comment (Chocolate-lightning) [reply .github/instructions/components.instructions.md]: Updated — the section now has two explicit paths: Bootstrap-based components include mds-* + Bootstrap classes; non-Bootstrap components use only mds-*. The example is labelled accordingly.

File: .github/instructions/stories-tests.instructions.md (hunk 2 above)
Comment (Chocolate-lightning) [reply .github/instructions/stories-tests.instructions.md]: Done — removed the inline code block. Both the meta object and story definition examples now point to components/button/Button.stories.tsx as the canonical reference.

File: .github/instructions/stories-tests.instructions.md (hunk 3 above)
Comment (Chocolate-lightning) [reply .github/instructions/stories-tests.instructions.md]: Same as before.

File: .github/instructions/stories-tests.instructions.md (hunk 4 above)
Comment (Chocolate-lightning) [reply .github/instructions/stories-tests.instructions.md:45]: Reframed as a minimum baseline with an explicit note to add additional edge cases and interactions specific to the component.

File: .github/instructions/stories-tests.instructions.md (hunk 5 above)
Comment (Chocolate-lightning) [reply .github/instructions/stories-tests.instructions.md]: Done — inline example replaced with a reference to components/button/Button.test.tsx.

File: .github/instructions/stories-tests.instructions.md (hunk 6 above)
Comment (Chocolate-lightning) [reply .github/instructions/stories-tests.instructions.md]: Similar to above.

File: .github/instructions/tokens.instructions.md (hunk 7 above)
Comment (Chocolate-lightning) [reply .github/instructions/tokens.instructions.md]: Changed to reflect correct URL. I think we can leave the GH issues and suggestions for now and have token requests come through ZH. We can refine the instructions later as our processes mature.

File: .github/instructions/tokens.instructions.md (hunk 8 above)
Comment (Chocolate-lightning) [reply .github/instructions/tokens.instructions.md]: Replaced with a reference to components/button/button.css

File: .github/instructions/tokens.instructions.md (hunk 9 above)
Comment (Chocolate-lightning) [reply .github/instructions/tokens.instructions.md]: Agreed — table row updated to "Renaming a token" (path removed) so the breaking change rule applies to all tokens regardless of location.

File: .github/instructions/tokens.instructions.md (hunk 11 above)
Comment (Chocolate-lightning) [reply .github/instructions/tokens.instructions.md]: Removed the standalone "Generated files — do not hand-edit" section

**Review by andrewgos — APPROVED** [review]
Thanks for actioning each of the items. Happy for this to merge 🚀
